src/Xamarin.Android.Build.Tasks/Tasks/FilterAssemblies.cs

```diff
@@ -0,0 +1,44 @@
+﻿using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using System.Collections.Generic;
+using System.IO;
+using System.Reflection.Metadata;
+using System.Reflection.PortableExecutable;
+
+namespace Xamarin.Android.Tasks
+{
+	/// <summary>
+	/// Filters a set of assemblies based on a given TargetFrameworkIdentifier
+	/// </summary>
+	public class FilterAssemblies : Task
+	{
+		[Required]
+		public string TargetFrameworkIdentifier { get; set; }
+
+		public ITaskItem [] InputAssemblies { get; set; }
+
+		[Output]
+		public ITaskItem [] OutputAssemblies { get; set; }
+
+		public override bool Execute ()
+		{
+			if (InputAssemblies == null)
+				return true;
+
+			var output = new List<ITaskItem> (InputAssemblies.Length);
+			foreach (var assemblyItem in InputAssemblies) {
+				using (var pe = new PEReader (File.OpenRead (assemblyItem.ItemSpec))) {
```

Comment:
I suppose this needs a `File.Exists()` check...

What *should* happen if a file doesn't exist?  Or is the file not existing a sign that something else went wrong?

Reply:
I added the logic we had in `<ResolveLibraryProjectImports/>`.